src/main/java/goral/psychotherapistoffice/domain/patient/PatientService.java

```diff
@@ -12,49 +14,68 @@
 public class PatientService {
 
 
-    private PatientJpaRepository patientJpaRepository;
+    private final PatientJpaRepository patientJpaRepository;
+    private final UserService userService;
 
-
-    public PatientService(PatientJpaRepository patientJpaRepository) {
+    public PatientService(PatientJpaRepository patientJpaRepository, UserService userService) {
         this.patientJpaRepository = patientJpaRepository;
+        this.userService = userService;
+
     }
 
-    public Optional<PatientDto> findPatientById(long id){
+    public Optional<PatientDto> findPatientById(long id) {
         return patientJpaRepository.findPatientById(id)
                 .map(PatientDtoMapper::map);
     }
 
-    public List <PatientDto> findBySurnameOrName(String keyword){
+    public List<PatientDto> findBySurnameOrName(String keyword) {
         return patientJpaRepository.findBySurnameContainsIgnoreCaseOrNameContainsIgnoreCase(keyword, keyword)
                 .stream()
                 .map(PatientDtoMapper::map)
                 .toList();
     }
 
-
-    public List<PatientDto>findAllPatients(){
+    public List<PatientDto> findAllPatients() {
         return patientJpaRepository.findAll()
                 .stream()
                 .map(PatientDtoMapper::map).toList();
     }
 
     @Transactional
-    public Patient addPatient(PatientDto patientDto){
+    public Patient addPatient(PatientDto patientDto) {
         Patient patientToSave = new Patient();
-        patientToSave.setNick(patientDto.getNick());
+        if (patientDto.getNick().isEmpty()) {
+            StringBuilder sB = new StringBuilder();
+            String tel = patientDto.getTelephone();
+            int n = tel.length();
+            for (int i = (n - 3);
+                 i < n;
+                 i++) {
+                char c = patientDto.getTelephone().charAt(i);
+                sB.append(c);
```

Comment:
we can replace lines 49-56 to one line:

```java
tel.substring(tel.length() - 3)
```

Reply:
how you find it??

Reply:
![image](https://github.com/GoralTomaszGorski/psychotherapist_office/assets/32451056/909b08b1-82ee-4854-8c0b-30e4f0dfb171)
this is not easy... :stuck_out_tongue_winking_eye:

Reply:
I saw this answer today, I will check it

Reply:
I tried to fix with using above code it is probably better but something wrong when I use substring method.
I fixed set free and true calendar therm.

Reply:
I fixed boolean set free false and true calendar therm during edit and add therm*